styx-flyte-client/src/main/java/com/spotify/styx/flyte/client/FlyteInputsUtils.java

```diff
@@ -101,19 +104,45 @@ private FlyteInputsUtils() {
     throw new UnsupportedOperationException("Can't find default value for launch plan input: " + key);
   }
 
-  static Literals.LiteralMap fillParameterInInputs(Interface.ParameterMap parameterMap, Map<String, String> extraDefaultInputs) {
+  static Literals.LiteralMap fillParameterInInputs(Interface.ParameterMap parameterMap, Map<String, String> userDefinedInputs, Map<String, String> styxVariables) {
+
+    // Validate that user defined inputs exist in the LaunchPlan
+    var paramsKeysInLowercase = parameterMap.getParametersMap().keySet().stream().map(String::toLowerCase).collect(toSet());
+    var lowercaseInputs = userDefinedInputs.keySet().stream().map(String::toLowerCase).collect(toSet());
+    if (!paramsKeysInLowercase.containsAll(lowercaseInputs)) {
+      var unMatchedInputKeys = userDefinedInputs.keySet().stream()
+          .filter(key -> !paramsKeysInLowercase.contains(key.toLowerCase()))
+          .collect(toList());
+      throw new UnsupportedOperationException("Inputs don't correspond with launch plans inputs:"
+                                              + " " + unMatchedInputKeys);
+    }
+
+
     var literalMapBuilder = Literals.LiteralMap.newBuilder();
-    var lowercaseExtraDefaultInputs = extraDefaultInputs.entrySet()
+
+    var combinedInputsLowerCase = FlyteInputsUtils.combineMapsCaseInsensitiveWithOrder(userDefinedInputs, styxVariables)
+        .entrySet()
         .stream()
-        .collect(ImmutableMap.toImmutableMap(x -> x.getKey().toLowerCase(), Map.Entry::getValue));
+        .collect(ImmutableMap.toImmutableMap(x -> x.getKey().toLowerCase(), Map.Entry::getValue))
+        ;
```

Comment:
I think that the `combineMapsCaseInsensitiveWithOrder` method should return the map as we want it so no need to do extra transformation here

Reply:
The keys are normalized toLowerCase to avoid any casing issues with the way they are defined in the parameterMap.
The code still expects a lowercased map